src/App.tsx

```diff
@@ -1,27 +1,86 @@
-import React from 'react';
+import React, { useState } from 'react';
+import 'bulma/css/bulma.min.css';
 import './App.scss';
+
+import { Good } from './types/Good';
+import { getAll, get5First, getRedGoods } from './api/goods';
+
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const throwError = () => {
+    setIsError(true);
+    setIsLoading(false);
+  };
+
+  const getAllGoods = () => {
```

Comment:
Consider wrapping your API calls in a useCallback to avoid unnecessary re-renders and memory leaks. Example: 'const getAllGoods = useCallback(async () => { ... }, []);' 🧹